libraries/freertos_plus/standard/https/include/types/iot_https_types.h

```diff
@@ -0,0 +1,544 @@
+ * 
+ * The user buffer is configured in #IotHttpsConnectionInfo_t.userBuffer and #IotHttpsRequestInfo_t.userBuffer.
+ * 
+ * The minimum size that the buffer must be configured to is indicated by requestUserBufferMinimumSize, 
+ * responseUserBufferMinimumSize, connectionUserBufferMinimumSize.
+ */
+typedef struct IotHttpsUserBuffer
+{
+    uint8_t * pBuffer;   /**< @brief Application provided buffer pointer. */
+    uint32_t bufferLen; /**< @brief The length of the application provided buffer. */
+} IotHttpsUserBuffer_t;
+
+/**
+ * @ingroup https_client_datatypes_paramstructs
+ * @brief HTTP/2 Stream place holder structure.
+ * 
+ * HTTP/2 has streams and there can be multiple request/response pairs going over a stream in parallel. 
+ * The best way to go about this might be to logically separate them. Then have an API to create new streams given 
+ * an existing connection. Internally to the library we can map streams to a given connection. A request can 
+ * then specify which stream to be associated with.  
+ * The mapping is: Connection --1:many--> Stream --1:many--> Request */
+typedef struct IotHttpsStream
+{
+    /** @brief Stream ID of the HTTPS request.
+     * In HTTP/2, we can send multiple requests in parallel on one connection if each of their discontinuos parts has
+     * a stream identifier. Stream 0 is the connection stream. Stream 1+ are the request/response streams. */
+    uint32_t streamId;                     
+    uint32_t priority;                     /**< @brief H2 stream priority, value between 1-256 (inclusive) */
+    uint32_t dependentStreamId;            /**< @brief H2 stream dependency on another stream */
+} IotHttpsStream_t;
+
+/**
+ * @ingroup https_client_datatypes_paramstructs
+ * @brief HTTP/2 Specific request information. 
+ * 
+ * This is a placeholder and to be expanded upon for future HTTP/2 support. 
+ */
+typedef struct IotpHttp2Request
+{
+    IotHttpsStream_t streamInfo; /* @brieff Required HTTP/2 Stream association. */
+
+    /* Other HTTPS2 needed items will be added here. */
+
+} IotHttp2RequestInfo_t;
+
+/**
+ * @ingroup https_client_datatypes_paramstructs
+ * @brief HTTPS Client synchronous request information.
+ * 
+ * @paramfor @ref https_client_function_initializerequest.
+ *
+ * This structure is configured in #IotHttpsRequestInfo_t.pSyncInfo.
+ * 
+ * A synchronous request will block until the response is fully received from the network. 
+ * This structure defines memory locations to store the response body. 
+ */
+typedef struct IotHttpsSyncRequestInfo
+{
+    /**
+     * @brief Pointer to the HTTP request entity body. 
+     * 
+     * This is the file or data we want to send. The data is separated from the headers for the flexibility to point to 
+     * an already established file elsewhere in memory.
+     * 
+     * Set this NULL if there is no request body.
+     */
+    uint8_t * pReqData;
+    uint32_t reqDataLen;                    /**< @brief length of the request data. */
```

Comment:
I do not want to use IotHttpsUserBuffer_t because the user buffer is specifically for context and headers. And putting IotHttpsUserBuffer_t here would make the structure nesting even deeper, which feels funny.

Reply:
https://github.com/aws/amazon-freertos/pull/829/files#diff-6a138290c6e456aac6f3e9f83858dab3R302